src/components/album/AlbumPreview.js

```diff
@@ -25,28 +29,51 @@ const AlbumPreview = ({
       return;
     }
     //extract the album id
-    const albumId = match[1];
+    const spotifyAlbumId = match[1];
+
     //  Spotify album URL for embedding
-    const spotifyAlbumUrl = `https://open.spotify.com/embed/album/${albumId}`;
+    const spotifyAlbumUrl = `https://open.spotify.com/embed/album/${spotifyAlbumId}`;
     setSpotifyEmbedUrl(spotifyAlbumUrl);
-  }, [spotifyUrl]);
 
+    // Fetch MongoDB album ID from  API
+    // AKOS: how would I get the id of the album in this album preview component to pass it later  to ReviewsList.js as a prop on line 68?
+    // mongo album id is not being set and instead of it it passes spotify id
+    axiosInstance
+      .get(`/albums/${albumId}`)
+      .then((response) => {
+        console.log(response.data.album._id);
+
+        setMongoAlbumId(response.data.album._id);
+      })
+      .catch((error) => {
+        console.error('Error fetching MongoDB album ID:', error);
+      });
+  }, [spotifyUrl]);
+  //console.log("Children:", children.type);
   return (
-    <CardContent style={{ marginTop: '1rem', width: '100%', display: 'flex' }}>
-      <iframe
-        title="Spotify Web Player"
-        src={spotifyEmbedUrl}
-        width={width}
-        height={height}
-        frameBorder={frameBorder}
-        style={{ borderRadius: 8, ...style, border: 'none' }}
-        allow={allow}
-        {...props}
-      />
-      {children && (
-        <div style={{ width: '100%', display: 'flex' }}>{children}</div>
-      )}
-    </CardContent>
+    <>
+      <CardContent
+        style={{ marginTop: '1rem', width: '100%', display: 'flex' }}
+      >
+        <iframe
+          title="Spotify Web Player"
+          src={spotifyEmbedUrl}
+          width={width}
+          height={height}
+          frameBorder={frameBorder}
+          style={{ borderRadius: 8, ...style, border: 'none' }}
+          allow={allow}
+          {...props}
+        />
+
+        {children && numOfReviews === 0 ? (
+          <p>No reviews available for this album</p>
+        ) : (
+          <div style={{ width: '100%', display: 'flex' }}>{children}</div>
+        )}
+      </CardContent>
+      <WriteReview />
```

Comment:
@akosasante Akos, here 
`We can (should) even make it a variable that we get from the props, that way the caller can be in charge of deciding how they want the album preview to be stacked with its children.`
 do you mean we should add this properties using  useState hook like on the screen?
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/103755090/20df178a-5ab2-47d9-90c4-2df9c10be456)
?

Reply:
No, not a `useState` hook variable. I mean that currently `AlbumPreview` takes these props: `spotifyUrl`, `style`, `wide`, `width`, `height`, `frameBorder`, `allow`, and `children` (children is a prop that _all_ React components have). 

These are currently mainly used to style the `iframe` element speciifcally. But we could add another prop that this component would accept called like `childrenStackDirection` or some better name. And then we can use that in the `CardContent` element's `style` attribute to set the `flexDirection` CSS value.

Reply:
That way, on the chat page where we want the iframe to stack horizontally with the chat component children, we can pass in `<AlbumPreview  childrenStackDirection='row'...etc />` and on the album modal where we want it to stack vertically with the review component, we can pass in `<AlbumPreview childrenStackDirection='column' ...etc />`

Reply:
@akosasante  I think I got it, just to make sure-normally we werepassing props from parent to a child-and here in this case we are destructuring these properties from props but we don't pass it  from anywhere, we only pass the first three from Album.js ,
```
 albumId,
  spotifyUrl,
  numOfReviews,
```
 so if the rest of the props are being assigned directly in props, they are not necessarily have to come from another component, right?
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/103755090/99138a0b-5e60-4f58-89c7-7054dcf12f99)
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/103755090/213850f8-af3e-4134-a8b4-e1100fe73fd4)

Reply:
The way this component is written, it requires the caller (parent component) to pass in: `albumId`, `spotifyUrl`, and `numOfReviews` otherwise those variables will be `undefined` which may or may not cause errors depending on how those variables are used.

For `style`, `wide`, `width`, `height`, `frameBorder`, and `allow`, the function that makes this component has default values set in the arguments for these props. So even if the parent component does not pass those in, they will be initialized with those default values.

This function component also uses the `rest` parameter (aka destructuring with `...`) to say "okay any other props that are passed in, just save it into one object variable called `props`", and it passes those into the `iframe`. So we should keep in mind that anything a parent component passes in, that is not explicitly named in those arguments, will end up getting passed to the `iframe` which may or may not be what we want.

My suggestion is to add an argument explicitly to that list for determining the flex layout, and then if it has a default value, that means anywhere where want to follow that default value, we do not need to explicitly name that prop in the parent component, but anywhere where we ant to override it we will need to pass that prop in from the parent component.